First draft of GitHub Skill-up content (#1)
carpentries/github-skills: episodes/managing_issues.md

[Milestones](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) are comparatively simple and a great place to start, especially if your efforts are focused on a single repository.
The focus will be on Milestones in this session but we recommend that you [learn more about Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects) if you want to take advantage of the full range of GitHub's features for project management.

Milestones are a way of organising issues on your project into smaller units of work (e.g. deliverables) that have deadlines and progress of which needs to be closely tracked (e.g. release management). 
They are good for managing time-bound, structured iterations and working in short sprints on batches of issues, which makes them a great fit for [the iterative process of lesson development recommended in Collaborative Lesson Development Training](https://carpentries.github.io/lesson-development-training/lesson-design.html). 

Here are some helpful features of Milestones:

> * Add a title and description to the Milestone, e.g. _"Issues to be addressed before October pilot workshop"_ or _"Accessibility Enhancement Drive"_.
> * The Milestone represents a subset of issues on your repository, e.g. the things you want to work on next or a set of issues that share a theme.
>   Issues are assigned to a Milestone in a similar way to adding a label.
> * The Milestone can be given a target completion date.
> * A progress bar indicates how close you are to completing the Milestone.
> * The order of issues listed in the Milestone can be adjusted e.g. to indicate task priority to yourself and your collaborators.

Reviewer: ```suggestion
* You can add a title and description to the Milestone, e.g. _"Issues to be addressed before October pilot workshop"_ or _"Accessibility Enhancement Drive"_.
* The Milestone represents a subset of issues on your repository, e.g. the things you want to work on next or a set of issues that share a theme.
  Issues are assigned to a Milestone in a similar way to adding a label.
* The Milestone can be given a target completion date.
* A progress bar indicates how close you are to completing the Milestone in terms of the number of completed issues verus the total number of issues.
* The order of issues listed in the Milestone can be adjusted manually by dragging the issue up and down the list, e.g. to indicate task priority to yourself and your collaborators.
```

Reviewer: Not sure if this extra text is helpful - perhaps for people who have not seen this before the extra explanations might be helpful. Will leave to @tobyhodges to decide.

::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

## Inviting Community Contributions
Lesson repositories are open source and publicly-visible, and you can expect some spontaneous contributions from the wider community.
However, in some circumstances, you may want to bring particular items to the attention of community members.

:::::::::::::::::::::::::::::::::::::::::::::::::::: spoiler

### For Carpentries Maintainers: Refer to CAC

> When you want or need to [refer an issue/contribution to the Curriculum Advisory Committee](https://carpentries.github.io/maintainer-onboarding/04-communicate-advisors.html) for your lesson, you should add the `status:refer to cac` label, and tag the CAC in the relevant issue/pull request discussion thread (e.g. `@datacarpentry/curriculum-advisors-image`).

Author: ```suggestion
When you want or need to [refer an issue/contribution to the Curriculum Advisory Committee](https://carpentries.github.io/maintainer-onboarding/04-communicate-advisors.html) for your lesson, you should add the `status:refer to cac` label, and mention the CAC in the relevant issue/pull request discussion thread (e.g. `@datacarpentry/curriculum-advisors-image`).
```

Adding this label will help members of the CAC more easily identify and browse through the items they need to discuss, increasing the likelihood that you will receive a timely response.

:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

### Help Wanted & Good First Issue
The `help wanted` and `good first issue` labels can be useful to draw attention from new contributors.
In particular, the `help wanted` label can be used to flag up issues that the Maintainers would particularly appreciate external support on.

> Issues with this label in any of The Carpentries official lesson repositories will appear in the listing on [the _Help Wanted Issues_ page of the website](https://carpentries.org/help-wanted-issues/), where new community members are often directed to find opportunities to contribute to a lesson repository.

Author: ```suggestion

```

The `good first issue` label is used to identify issues that do not require in-depth knowledge of the project and its infrastructure, etc.
Newcomers to the project can use this label to find opportunities to make their first contribution.
GitHub provides a view of all of a repository's issues with the `good first issue` label on the `/contribute` page for that repository, e.g. https://github.com/github/docs/contribute

> ::: callout

Reviewer 2: ```suggestion
::: spoiler
```

Different tools will work best for different teams of collaborators, but making sure you use any tool to plan and manage work on your lesson development project is better than making it up as you go along.

### Milestones
Carpentries lessons are developed in GitHub so it is convenient (but not a must) to use its features to keep track of different components of the project and their current status. 

GitHub offers two features for task/project management: _Projects_ and _Milestones_.
Projects offer a more advanced set of functions and are particularly useful when you need to work on related tasks spread across multiple repositories.
[Milestones](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) are comparatively simple and a great place to start, especially if your efforts are focused on a single repository.
The focus will be on Milestones in this session but we recommend that you [learn more about Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects) if you want to take advantage of the full range of GitHub's features for project management.

> Milestones are a way of organising issues on your project into smaller units of work (e.g. deliverables) that have deadlines and progress of which needs to be closely tracked (e.g. release management). 

Reviewer 3: ```suggestion
Milestones are a way of organising issues on your project into smaller units of work (e.g. deliverables) that have deadlines and the progress of which needs to be closely tracked (e.g. release management). 
```

> *We are not direct maintainers of the lesson, but these issues are important for us to fix before teaching the next edition of the workshop, because we don't want to run into them again. 
> Of course other users of the lesson would also benefit from the issues being fixed.*

> *Therefore, we organised a one-day sprint in which we gathered together colleagues from different teams to work on improving the lesson. 
> By all working together on one day, together in the same room, we were able to focus and get a lot done in a short amount of time, while not being distracted by the hundreds of other things we are all working on. 
> Of course, we contacted the lesson maintainers to make sure that our efforts would be welcome by opening [an issue in their lesson repository](https://github.com/carpentries-incubator/python-intermediate-development/issues/343).*

> *We used a Microsoft Planner board to organise our work and keep track of what needs to happen:*

> ![Microsoft Planner project management board](https://github.com/carpentries/lesson-development-training/assets/9945255/7705b6a8-2387-4e00-a8b8-1eaa2a512260){

> alt='Microsoft Planner project management board displaying TO-DO tasks, tasks being worked on, tasks ready for internal review and completed tasks'

Reviewer 3: ```suggestion
alt='Screenshot of Microsoft Planner project management board displaying TO-DO tasks, tasks being worked on, tasks ready for internal review and completed tasks'
```

> *In this way, we could adapt the lesson to our own needs and contribute to the lesson in The Carpentries Incubator in a smooth way that does not take away focus on other projects of me and my colleagues."*

Renata Curty, Research Facilitator for Social Sciences at UCSB, describes her group's approach:

> *Our group uses issues and labels and sets milestones in GitHub to organize our lesson development work.*
> *We also use a spreadsheet to track higher-level decisions, tasks, and assignees and record ideas and discussion points in our meeting notes before converting them into more detailed issues.*
> *We have been meeting regularly (weekly or biweekly) to maintain momentum, which helps us to make small but steady progress.*
> *We are currently working on a modularized version of the workshop that will allow instructors to choose what to teach based on the time, cohort needs, and learners’ entry knowledge.*

::::::::::::::::::::::::::::::::::::::

:::::::::::::::::::::::::::::: callout

### Referencing issues
It can be helpful to refer to issues and pull requests in conversations elsewhere in a repository.

> Every issue and PR has a unique number (based on the order in which they were created).These numbers can be used, preceded by `#`, to reference a given issue or PR.

Reviewer 3: ```suggestion
Every issue and PR has a unique number (based on the order in which they were created). These numbers can be used, preceded by `#`, to reference a given issue or PR.
```

Additionally, pull requests (and commit messages) that include particular phrases will trigger the automatic closure of an issue:

- "closes"
- "fixes"
- "resolves"

> One of the above words followed with the `#` and the number of an issue, e.g. _"fixes #34"_, included in the description of a pull request (the first comment in the PR discussion thread) will tell GitHub that the issue can be automatically closed if/when the pull request is merged.

Reviewer 3: ```suggestion
One of the above words followed with the `#` and the number of an issue, e.g. _"fixes #34"_, included in the description of a pull request (the first comment in the PR discussion thread) will tell GitHub that the issue can be automatically closed if the pull request is merged.
```
